### ISSUE TITLE
monero-gui-v0.15.0.1 "E Device not found in registry: 'Trezor'. Known devices:"

### ISSUE THREAD

**Issue by redmac68** [monero-project/monero-gui#2479]
OS: Linux Mint Debian Edition 3 (LMDE3)
-monero-gui-v14.1.0 running correctly with hardware wallet using trezor model T
-installed monero-gui-v0.15.0.1 to new folder
-ran monero-wallet-gui
-get message on gui "Couldn't open wallet:device not found: Trezor"

-terminal shows:
neil@xps13-mint-neil ~/monero-gui-v0.15.0.1 $ ./monero-wallet-gui
2019-11-24 11:20:33.118	W app startd (log: /home/neil/.bitmonero/monero-wallet-gui.log)
2019-11-24 11:20:33.119	W Qt:5.9.7 GUI:v0.15.0.1 | screen: 1920x1080 - dpi: 96 - ratio:1.2996
2019-11-24 11:20:35.594	W qrc:/qt-project.org/imports/QtQuick/Controls/ApplicationWindow.qml:240:9: QML ContentItem: Binding loop detected for property "implicitWidth"
2019-11-24 11:20:50.346	I [PARSE URI] regex not matched for uri: ^((.*?)://)?(\[(.*)\](:(\d+))?)(.*)?
2019-11-24 11:20:50.347	I Generating SSL certificate
2019-11-24 11:20:50.350	I ringdb path set to /home/neil/.shared-ringdb
2019-11-24 11:20:50.525	W Account on device. Initing device...
**2019-11-24 11:20:50.526	E Device not found in registry: 'Trezor'. Known devices: 
2019-11-24 11:20:50.526	E  - Ledger
2019-11-24 11:20:50.526	E  - default
2019-11-24 11:20:50.526	E Error opening wallet: device not found: Trezor
2019-11-24 11:20:50.530	E Error opening wallet with password:  device not found: Trezor**

Double checked that monero-gui-v14.1.0 still works correctly with Trezor.

**Comment by xiphon** [comment]
Seems release binaries are accidentally compiled without Trezor support.
Would you like to test the new ones?

**Comment by redmac68** [comment]
I would be honoured. Thanks.

**Comment by xiphon** [comment]
https://anonfile.com/WeB79fB8nf/monero-wallet-gui.tar_xz
```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

3e936ee7b493341b7db2dce8f1c82220e05692752e30062f834ea449b97e1294  monero-wallet-gui.tar.xz
#
# signed by Xiphon
#
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCAAdFiEEj3eWTbujJBmNeKYDvXLsbD8YfGcFAl3ahroACgkQvXLsbD8Y
fGdLBwv/VIG5eA4m1v3qnZ++lsJsRaTrc+6CLxiO64ChBN9K1CAKW7xX6B1MGkr2
q8nRU7mbqLfYic+scPcfQ1LHAogCGgPptNa3OUOIgJQRQyccCFCcKO/9Xey79Sqh
HJ/GYdG3J4WmHBcJf5PVwJxpuaH8fQynap8KgNIyEYNkKnLwf9hMguC4SKVnXctz
v2ObAhA4Iv56jVpBCEH6mRZSPuduj3MDqlFioP4EL6rkhnYYrEZvTXQr4CorftiB
aMPLAGM9x/Whi/aUmCVKhP03u0vQrr1XEvj5AzVM+xbGnhtALV7WA4pgdY/snJtp
ujdELnlCFEXEc9oD8b2RSI02AokBGFAn2M6Gy4vgP0qR+ZWwXCGkqRCmckv924tw
+dlCxljaFG7wgrpwcUMumGyc+hL+q+FC1X7vOa4C/fBjBwygGDg1sZLxU/97F/IB
lyh0Ld2ysYxj6wjNW9s78kuz3vxoklOatPwebVOBlzpAakRUb2Cceai8tRmc9SYp
RET3UL3y
=1HWl
-----END PGP SIGNATURE-----
```

**Comment by redmac68** [comment]
Hi, that looks good .. Trezor device is found. 
Testing limited as I am using a remote node and have "Network status: Wrong version" and "Connected daemon is not compatible with GUI. Please upgrade or connect to another daemon".

 I have tried the remote node "uwillrunanodesoon.moneroworld.com:18089", but I still get the same errors. 

Is there any trusted remote node running 15.0.1 daemon or does this have to wait for 3oth Nov?

**Comment by selsta** [comment]
Please try node.xmr.to:18081

**Comment by redmac68** [comment]
Hi, thanks again. That node works. 

I tried to send and got an error
"Can't create transaction: unexpected error: Call method failed"

.. but I guess this is to be expected until all nodes are updated. If there is something else that I should check, let me know.

**Comment by selsta** [comment]
Please install the beta firmware from beta-wallet.trezor.io and try again.

**Comment by redmac68** [comment]
Hi, the Trezor has the latest firmware version already installed (2.1.8). 
I don't see any other version available.
The message in a terminal window when the send fails is like:

2019-11-24 14:50:27.702	I Decrypted payment ID: <0000000000000000>
2019-11-24 14:50:27.760	E Failed to invoke http request to  /call/21, wrong response code: 400 Response Body: {"error":"session not found"}
2019-11-24 14:50:27.766	E Can't create transaction:  unexpected error: Call method failed

I re-tested sending a small amount to my own address .. and this was successful.

**Comment by schulzemic** [comment]
I also tried this now. Updated the firmware to 2.1.8, and sent to my own subaddress, using my own remote note. It worked, thank you!

**Comment by selsta** [comment]
We will release a v0.15.0.2 point release in the next days to fix this. In the meantime the binary @xiphon posted above can be used.

**Comment by xiphon** [comment]
Removed the link to binary i posted above. It was uploaded just for testing purposes (to be sure that Trezor support issue is fixed). Is not a good idea to let users download random binaries,

Please wait for v0.15.0.2 point release in a few days.

@redmac68 @schulzemic thanks for testing.

**Comment by redmac68** [comment]
@xiphon @selsta @schulzemic thanks for your support .... beyond awesome.

**Comment by selsta** [comment]
+resolved

**Comment by redmac68** [comment]
Monero Gui Linux x64 v 0.15.0.4 now gives message "Couldn't open wallet: device not found: Trezor"

This from terminal window ..

2020-03-20 14:18:02.868	W app startd (log: /home/neil/.bitmonero/monero-wallet-gui.log)
2020-03-20 14:18:02.869	W Qt:5.9.7 GUI:v0.15.0.4 | screen: 1920x1080 - dpi: 96 - ratio:1.2996
2020-03-20 14:18:07.090	W qrc:/qt-project.org/imports/QtQuick/Controls/ApplicationWindow.qml:240:9: QML ContentItem: Binding loop detected for property "implicitWidth"
2020-03-20 14:18:38.191	I [PARSE URI] regex not matched for uri: ^((.*?)://)?(\[(.*)\](:(\d+))?)(.*)?
2020-03-20 14:18:38.191	I Generating SSL certificate
2020-03-20 14:18:38.192	I ringdb path set to /home/neil/.shared-ringdb
2020-03-20 14:18:38.301	W Account on device. Initing device...
2020-03-20 14:18:38.301	E Device not found in registry: 'Trezor'. Known devices: 
2020-03-20 14:18:38.301	E  - Ledger
2020-03-20 14:18:38.301	E  - default
2020-03-20 14:18:38.306	E Error opening wallet: device not found: Trezor
2020-03-20 14:18:38.315	E Error opening wallet with password:  device not found: Trezor

Can someone please check/re-open?

**Comment by selsta** [comment]
@redmac68 Known issue, will be fixed with v0.15.0.5. I’ve also added the Trezor dependency to the README so that this doesn’t happen again.

https://github.com/monero-project/monero-gui/pull/2796

**Comment by redmac68** [comment]
Great news, thanks again.
